sormas-ui/src/main/java/de/symeda/sormas/ui/utils/AbstractEditForm.java

```diff
@@ -323,6 +323,9 @@ protected void setVisible(boolean visible, String... fieldOrPropertyIds) {
 		for (String propertyId : fieldOrPropertyIds) {
 			if (!visible || isVisibleAllowed(propertyId)) {
 				getField(propertyId).setVisible(visible);
+				if (visible == false) {
```

Comment:
Even though it often is a good idea to solve problems in a generic way , I fear that this breaks some functionalities.
I can not provide concrete examples right now, but I think there are fields that should be hidden but still hold values. This might, for example be necessary for the SurvNet interface or reduced features.
That's why I suggest to clear fields via change listeners instead.

Reply:
Furthermore, this change often seems to lead to crashes.
For example, when trying to create this case:

<img width="402" alt="Error2" src="https://user-images.githubusercontent.com/67001822/159670351-10fec374-3419-4d19-a40f-0366a6db2c60.PNG">


```com.vaadin.server.ServerRpcManager$RpcInvocationException: Unable to invoke method click in com.vaadin.shared.ui.button.ButtonServerRpc
at com.vaadin.server.ServerRpcManager.applyInvocation(ServerRpcManager.java:159)
at com.vaadin.server.ServerRpcManager.applyInvocation(ServerRpcManager.java:116)
at com.vaadin.server.communication.ServerRpcHandler.handleInvocation(ServerRpcHandler.java:442)
at com.vaadin.server.communication.ServerRpcHandler.handleInvocations(ServerRpcHandler.java:407)
at com.vaadin.server.communication.ServerRpcHandler.handleRpc(ServerRpcHandler.java:275)
at com.vaadin.server.communication.UidlRequestHandler.synchronizedHandleRequest(UidlRequestHandler.java:91)
at com.vaadin.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:40)
at com.vaadin.server.VaadinService.handleRequest(VaadinService.java:1637)
at com.vaadin.server.VaadinServlet.service(VaadinServlet.java:464)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:750)
at org.apache.catalina.core.StandardWrapper.service(StandardWrapper.java:1637)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:331)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:211)
at de.symeda.sormas.ui.SessionFilter.lambda$doFilter$2(SessionFilter.java:73)
at de.symeda.sormas.ui.SessionFilterBean.doFilter(SessionFilterBean.java:37)
at jdk.internal.reflect.GeneratedMethodAccessor457.invoke(Unknown Source)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.glassfish.ejb.security.application.EJBSecurityManager.runMethod(EJBSecurityManager.java:588)
at org.glassfish.ejb.security.application.EJBSecurityManager.invoke(EJBSecurityManager.java:408)
at com.sun.ejb.containers.BaseContainer.invokeBeanMethod(BaseContainer.java:4835)
at com.sun.ejb.EjbInvocation.invokeBeanMethod(EjbInvocation.java:665)
at com.sun.ejb.containers.interceptors.AroundInvokeChainImpl.invokeNext(InterceptorManager.java:834)
at com.sun.ejb.EjbInvocation.proceed(EjbInvocation.java:615)
at com.sun.ejb.containers.interceptors.SystemInterceptorProxy.doCall(SystemInterceptorProxy.java:163)
at com.sun.ejb.containers.interceptors.SystemInterceptorProxy.aroundInvoke(SystemInterceptorProxy.java:140)
at jdk.internal.reflect.GeneratedMethodAccessor312.invoke(Unknown Source)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at com.sun.ejb.containers.interceptors.AroundInvokeInterceptor.intercept(InterceptorManager.java:888)
at com.sun.ejb.containers.interceptors.AroundInvokeChainImpl.invokeNext(InterceptorManager.java:833)
at com.sun.ejb.EjbInvocation.proceed(EjbInvocation.java:615)
at org.jboss.weld.module.ejb.AbstractEJBRequestScopeActivationInterceptor.aroundInvoke(AbstractEJBRequestScopeActivationInterceptor.java:72)
at org.jboss.weld.module.ejb.SessionBeanInterceptor.aroundInvoke(SessionBeanInterceptor.java:52)
at jdk.internal.reflect.GeneratedMethodAccessor278.invoke(Unknown Source)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at com.sun.ejb.containers.interceptors.AroundInvokeInterceptor.intercept(InterceptorManager.java:888)
at com.sun.ejb.containers.interceptors.AroundInvokeChainImpl.invokeNext(InterceptorManager.java:833)
at com.sun.ejb.containers.interceptors.InterceptorManager.intercept(InterceptorManager.java:375)
at com.sun.ejb.containers.BaseContainer.__intercept(BaseContainer.java:4807)
at com.sun.ejb.containers.BaseContainer.intercept(BaseContainer.java:4795)
at com.sun.ejb.containers.EJBLocalObjectInvocationHandler.invoke(EJBLocalObjectInvocationHandler.java:212)
at com.sun.ejb.containers.EJBLocalObjectInvocationHandlerDelegate.invoke(EJBLocalObjectInvocationHandlerDelegate.java:90)
at com.sun.proxy.$Proxy862.doFilter(Unknown Source)
at de.symeda.sormas.ui.__EJB31_Generated__SessionFilterBean__Intf____Bean__.doFilter(Unknown Source)
at de.symeda.sormas.ui.SessionFilter.doFilter(SessionFilter.java:66)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:253)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:211)
at de.symeda.sormas.ui.Slf4jMdcFilter.doFilter(Slf4jMdcFilter.java:29)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:253)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:211)
at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:257)
at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:160)
at org.apache.catalina.core.StandardPipeline.doInvoke(StandardPipeline.java:757)
at org.apache.catalina.core.StandardPipeline.doChainInvoke(StandardPipeline.java:581)
at com.sun.enterprise.web.WebPipeline.invoke(WebPipeline.java:97)
at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:158)
at org.apache.catalina.connector.CoyoteAdapter.doService(CoyoteAdapter.java:371)
at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:238)
at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:520)
at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:217)
at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:182)
at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:156)
at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:218)
at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:95)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:260)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:177)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:109)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:88)
at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:53)
at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:524)
at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:89)
at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:94)
at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:33)
at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:114)
at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:569)
at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:549)
at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.reflect.InvocationTargetException
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at com.vaadin.server.ServerRpcManager.applyInvocation(ServerRpcManager.java:155)
... 78 more
Caused by: com.vaadin.event.ListenerMethod$MethodException: Invocation of method buttonClick in de.symeda.sormas.ui.utils.CommitDiscardWrapperComponent$1 failed.
at com.vaadin.event.ListenerMethod.receiveEvent(ListenerMethod.java:726)
at com.vaadin.event.EventRouter.fireEvent(EventRouter.java:399)
at com.vaadin.event.EventRouter.fireEvent(EventRouter.java:363)
at com.vaadin.server.AbstractClientConnector.fireEvent(AbstractClientConnector.java:1190)
at com.vaadin.ui.Button.fireClick(Button.java:384)
at com.vaadin.ui.Button$1.click(Button.java:57)
... 83 more
Caused by: com.vaadin.event.ListenerMethod$MethodException: Invocation of method valueChange in de.symeda.sormas.ui.caze.CaseDataForm$$Lambda$2790/0x00000008022e1440 failed.
at com.vaadin.event.ListenerMethod.receiveEvent(ListenerMethod.java:726)
at com.vaadin.event.EventRouter.fireEvent(EventRouter.java:399)
at com.vaadin.event.EventRouter.fireEvent(EventRouter.java:363)
at com.vaadin.server.AbstractClientConnector.fireEvent(AbstractClientConnector.java:1190)
at com.vaadin.v7.ui.AbstractField.fireValueChange(AbstractField.java:1134)
at com.vaadin.v7.ui.AbstractField.setValue(AbstractField.java:545)
at com.vaadin.v7.ui.AbstractField.setValue(AbstractField.java:442)
at com.vaadin.v7.ui.AbstractField.setValue(AbstractField.java:427)
at de.symeda.sormas.ui.utils.AbstractEditForm.setValue(AbstractEditForm.java:100)
at de.symeda.sormas.ui.caze.CaseDataForm.setValue(CaseDataForm.java:1656)
at de.symeda.sormas.ui.caze.CaseController.getCaseDataEditComponent(CaseController.java:844)
at de.symeda.sormas.ui.caze.CaseDataView.initView(CaseDataView.java:93)
at de.symeda.sormas.ui.caze.AbstractCaseView.enter(AbstractCaseView.java:240)
at com.vaadin.navigator.Navigator.performNavigateTo(Navigator.java:778)
at com.vaadin.navigator.Navigator.lambda$navigateTo$9a874efd$1(Navigator.java:702)
at com.vaadin.navigator.ViewBeforeLeaveEvent.navigate(ViewBeforeLeaveEvent.java:54)
at com.vaadin.navigator.View.beforeLeave(View.java:79)
at com.vaadin.navigator.Navigator.runAfterLeaveConfirmation(Navigator.java:730)
at com.vaadin.navigator.Navigator.navigateTo(Navigator.java:701)
at com.vaadin.navigator.Navigator.navigateTo(Navigator.java:678)
at de.symeda.sormas.ui.caze.CaseController.navigateToView(CaseController.java:451)
at de.symeda.sormas.ui.caze.CaseController.navigateToView(CaseController.java:437)
at de.symeda.sormas.ui.caze.CaseController.lambda$selectOrCreateCase$10(CaseController.java:789)
at de.symeda.sormas.ui.caze.CaseController.selectOrCreateCase(CaseController.java:829)
at de.symeda.sormas.ui.caze.CaseController.selectOrCreateCase(CaseController.java:785)
at de.symeda.sormas.ui.caze.CaseController.lambda$getCaseCreateComponent$8(CaseController.java:771)
at de.symeda.sormas.ui.person.PersonController.selectOrCreatePerson(PersonController.java:146)
at de.symeda.sormas.ui.person.PersonController.selectOrCreatePerson(PersonController.java:84)
at de.symeda.sormas.ui.caze.CaseController.lambda$getCaseCreateComponent$9(CaseController.java:765)
at de.symeda.sormas.ui.utils.CommitDiscardWrapperComponent.onCommit(CommitDiscardWrapperComponent.java:668)
at de.symeda.sormas.ui.utils.CommitDiscardWrapperComponent.doCommit(CommitDiscardWrapperComponent.java:515)
at de.symeda.sormas.ui.utils.CommitDiscardWrapperComponent.commit(CommitDiscardWrapperComponent.java:450)
at de.symeda.sormas.ui.utils.CommitDiscardWrapperComponent.commitAndHandle(CommitDiscardWrapperComponent.java:545)
at de.symeda.sormas.ui.utils.CommitDiscardWrapperComponent$1.buttonClick(CommitDiscardWrapperComponent.java:371)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at com.vaadin.event.ListenerMethod.receiveEvent(ListenerMethod.java:706)
... 88 more
Caused by: com.vaadin.v7.data.Property$ReadOnlyException
at com.vaadin.v7.ui.AbstractField.setValue(AbstractField.java:467)
at com.vaadin.v7.ui.AbstractField.setValue(AbstractField.java:442)
at com.vaadin.v7.ui.AbstractField.setValue(AbstractField.java:427)
at com.vaadin.v7.ui.AbstractTextField.setValue(AbstractTextField.java:458)
at com.vaadin.v7.ui.TextField.clear(TextField.java:151)
at de.symeda.sormas.ui.utils.AbstractEditForm.setVisible(AbstractEditForm.java:327)
at de.symeda.sormas.ui.caze.CaseDataForm.lambda$addFields$3577bca6$1(CaseDataForm.java:1325)
at jdk.internal.reflect.GeneratedMethodAccessor660.invoke(Unknown Source)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at com.vaadin.event.ListenerMethod.receiveEvent(ListenerMethod.java:706)
... 126 more
]]

[2022-03-23T10:36:35.818+0100] [Payara 5.2021.10] [INFORMATION] [] [] [tid: _ThreadID=96 _ThreadName=http-thread-pool::http-listener-1(2)] [timeMillis: 1648028195818] [levelValue: 800] [[
E103635.816|     > Unable to invoke method click in com.vaadin.shared.ui.button.ButtonServerRpc (de.symeda.sormas.ui.SormasErrorHandler.java:80)
com.vaadin.server.ServerRpcManager$RpcInvocationException: Unable to invoke method click in com.vaadin.shared.ui.button.ButtonServerRpc
at com.vaadin.server.ServerRpcManager.applyInvocation(ServerRpcManager.java:159)
at com.vaadin.server.ServerRpcManager.applyInvocation(ServerRpcManager.java:116)
at com.vaadin.server.communication.ServerRpcHandler.handleInvocation(ServerRpcHandler.java:442)
at com.vaadin.server.communication.ServerRpcHandler.handleInvocations(ServerRpcHandler.java:407)
at com.vaadin.server.communication.ServerRpcHandler.handleRpc(ServerRpcHandler.java:275)
at com.vaadin.server.communication.UidlRequestHandler.synchronizedHandleRequest(UidlRequestHandler.java:91)
at com.vaadin.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:40)
at com.vaadin.server.VaadinService.handleRequest(VaadinService.java:1637)
at com.vaadin.server.VaadinServlet.service(VaadinServlet.java:464)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:750)
at org.apache.catalina.core.StandardWrapper.service(StandardWrapper.java:1637)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:331)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:211)
at de.symeda.sormas.ui.SessionFilter.lambda$doFilter$2(SessionFilter.java:73)
at de.symeda.sormas.ui.SessionFilterBean.doFilter(SessionFilterBean.java:37)
at jdk.internal.reflect.GeneratedMethodAccessor457.invoke(Unknown Source)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.glassfish.ejb.security.application.EJBSecurityManager.runMethod(EJBSecurityManager.java:588)
at org.glassfish.ejb.security.application.EJBSecurityManager.invoke(EJBSecurityManager.java:408)
at com.sun.ejb.containers.BaseContainer.invokeBeanMethod(BaseContainer.java:4835)
at com.sun.ejb.EjbInvocation.invokeBeanMethod(EjbInvocation.java:665)
at com.sun.ejb.containers.interceptors.AroundInvokeChainImpl.invokeNext(InterceptorManager.java:834)
at com.sun.ejb.EjbInvocation.proceed(EjbInvocation.java:615)
at com.sun.ejb.containers.interceptors.SystemInterceptorProxy.doCall(SystemInterceptorProxy.java:163)
at com.sun.ejb.containers.interceptors.SystemInterceptorProxy.aroundInvoke(SystemInterceptorProxy.java:140)
at jdk.internal.reflect.GeneratedMethodAccessor312.invoke(Unknown Source)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at com.sun.ejb.containers.interceptors.AroundInvokeInterceptor.intercept(InterceptorManager.java:888)
at com.sun.ejb.containers.interceptors.AroundInvokeChainImpl.invokeNext(InterceptorManager.java:833)
at com.sun.ejb.EjbInvocation.proceed(EjbInvocation.java:615)
at org.jboss.weld.module.ejb.AbstractEJBRequestScopeActivationInterceptor.aroundInvoke(AbstractEJBRequestScopeActivationInterceptor.java:72)
at org.jboss.weld.module.ejb.SessionBeanInterceptor.aroundInvoke(SessionBeanInterceptor.java:52)
at jdk.internal.reflect.GeneratedMethodAccessor278.invoke(Unknown Source)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at com.sun.ejb.containers.interceptors.AroundInvokeInterceptor.intercept(InterceptorManager.java:888)
at com.sun.ejb.containers.interceptors.AroundInvokeChainImpl.invokeNext(InterceptorManager.java:833)
at com.sun.ejb.containers.interceptors.InterceptorManager.intercept(InterceptorManager.java:375)
at com.sun.ejb.containers.BaseContainer.__intercept(BaseContainer.java:4807)
at com.sun.ejb.containers.BaseContainer.intercept(BaseContainer.java:4795)
at com.sun.ejb.containers.EJBLocalObjectInvocationHandler.invoke(EJBLocalObjectInvocationHandler.java:212)
at com.sun.ejb.containers.EJBLocalObjectInvocationHandlerDelegate.invoke(EJBLocalObjectInvocationHandlerDelegate.java:90)
at com.sun.proxy.$Proxy862.doFilter(Unknown Source)
at de.symeda.sormas.ui.__EJB31_Generated__SessionFilterBean__Intf____Bean__.doFilter(Unknown Source)
at de.symeda.sormas.ui.SessionFilter.doFilter(SessionFilter.java:66)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:253)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:211)
at de.symeda.sormas.ui.Slf4jMdcFilter.doFilter(Slf4jMdcFilter.java:29)
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:253)
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:211)
at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:257)
at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:160)
at org.apache.catalina.core.StandardPipeline.doInvoke(StandardPipeline.java:757)
at org.apache.catalina.core.StandardPipeline.doChainInvoke(StandardPipeline.java:581)
at com.sun.enterprise.web.WebPipeline.invoke(WebPipeline.java:97)
at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:158)
at org.apache.catalina.connector.CoyoteAdapter.doService(CoyoteAdapter.java:371)
at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:238)
at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:520)
at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:217)
at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:182)
at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:156)
at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:218)
at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:95)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:260)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:177)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:109)
at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:88)
at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:53)
at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:524)
at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:89)
at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:94)
at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:33)
at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:114)
at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:569)
at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:549)
at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.reflect.InvocationTargetException: null
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at com.vaadin.server.ServerRpcManager.applyInvocation(ServerRpcManager.java:155)
... 78 common frames omitted
Caused by: com.vaadin.event.ListenerMethod$MethodException: Invocation of method buttonClick in de.symeda.sormas.ui.utils.CommitDiscardWrapperComponent$1 failed.
at com.vaadin.event.ListenerMethod.receiveEvent(ListenerMethod.java:726)
at com.vaadin.event.EventRouter.fireEvent(EventRouter.java:399)
at com.vaadin.event.EventRouter.fireEvent(EventRouter.java:363)
at com.vaadin.server.AbstractClientConnector.fireEvent(AbstractClientConnector.java:1190)
at com.vaadin.ui.Button.fireClick(Button.java:384)
at com.vaadin.ui.Button$1.click(Button.java:57)
... 83 common frames omitted
Caused by: com.vaadin.event.ListenerMethod$MethodException: Invocation of method valueChange in de.symeda.sormas.ui.caze.CaseDataForm$$Lambda$2790/0x00000008022e1440 failed.
at com.vaadin.event.ListenerMethod.receiveEvent(ListenerMethod.java:726)
at com.vaadin.event.EventRouter.fireEvent(EventRouter.java:399)
at com.vaadin.event.EventRouter.fireEvent(EventRouter.java:363)
at com.vaadin.server.AbstractClientConnector.fireEvent(AbstractClientConnector.java:1190)
at com.vaadin.v7.ui.AbstractField.fireValueChange(AbstractField.java:1134)
at com.vaadin.v7.ui.AbstractField.setValue(AbstractField.java:545)
at com.vaadin.v7.ui.AbstractField.setValue(AbstractField.java:442)
at com.vaadin.v7.ui.AbstractField.setValue(AbstractField.java:427)
at de.symeda.sormas.ui.utils.AbstractEditForm.setValue(AbstractEditForm.java:100)
at de.symeda.sormas.ui.caze.CaseDataForm.setValue(CaseDataForm.java:1656)
at de.symeda.sormas.ui.caze.CaseController.getCaseDataEditComponent(CaseController.java:844)
at de.symeda.sormas.ui.caze.CaseDataView.initView(CaseDataView.java:93)
at de.symeda.sormas.ui.caze.AbstractCaseView.enter(AbstractCaseView.java:240)
at com.vaadin.navigator.Navigator.performNavigateTo(Navigator.java:778)
at com.vaadin.navigator.Navigator.lambda$navigateTo$9a874efd$1(Navigator.java:702)
at com.vaadin.navigator.ViewBeforeLeaveEvent.navigate(ViewBeforeLeaveEvent.java:54)
at com.vaadin.navigator.View.beforeLeave(View.java:79)
at com.vaadin.navigator.Navigator.runAfterLeaveConfirmation(Navigator.java:730)
at com.vaadin.navigator.Navigator.navigateTo(Navigator.java:701)
at com.vaadin.navigator.Navigator.navigateTo(Navigator.java:678)
at de.symeda.sormas.ui.caze.CaseController.navigateToView(CaseController.java:451)
at de.symeda.sormas.ui.caze.CaseController.navigateToView(CaseController.java:437)
at de.symeda.sormas.ui.caze.CaseController.lambda$selectOrCreateCase$10(CaseController.java:789)
at de.symeda.sormas.ui.caze.CaseController.selectOrCreateCase(CaseController.java:829)
at de.symeda.sormas.ui.caze.CaseController.selectOrCreateCase(CaseController.java:785)
at de.symeda.sormas.ui.caze.CaseController.lambda$getCaseCreateComponent$8(CaseController.java:771)
at de.symeda.sormas.ui.person.PersonController.selectOrCreatePerson(PersonController.java:146)
at de.symeda.sormas.ui.person.PersonController.selectOrCreatePerson(PersonController.java:84)
at de.symeda.sormas.ui.caze.CaseController.lambda$getCaseCreateComponent$9(CaseController.java:765)
at de.symeda.sormas.ui.utils.CommitDiscardWrapperComponent.onCommit(CommitDiscardWrapperComponent.java:668)
at de.symeda.sormas.ui.utils.CommitDiscardWrapperComponent.doCommit(CommitDiscardWrapperComponent.java:515)
at de.symeda.sormas.ui.utils.CommitDiscardWrapperComponent.commit(CommitDiscardWrapperComponent.java:450)
at de.symeda.sormas.ui.utils.CommitDiscardWrapperComponent.commitAndHandle(CommitDiscardWrapperComponent.java:545)
at de.symeda.sormas.ui.utils.CommitDiscardWrapperComponent$1.buttonClick(CommitDiscardWrapperComponent.java:371)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at com.vaadin.event.ListenerMethod.receiveEvent(ListenerMethod.java:706)
... 88 common frames omitted
Caused by: com.vaadin.v7.data.Property$ReadOnlyException: null
at com.vaadin.v7.ui.AbstractField.setValue(AbstractField.java:467)
at com.vaadin.v7.ui.AbstractField.setValue(AbstractField.java:442)
at com.vaadin.v7.ui.AbstractField.setValue(AbstractField.java:427)
at com.vaadin.v7.ui.AbstractTextField.setValue(AbstractTextField.java:458)
at com.vaadin.v7.ui.TextField.clear(TextField.java:151)
at de.symeda.sormas.ui.utils.AbstractEditForm.setVisible(AbstractEditForm.java:327)
at de.symeda.sormas.ui.caze.CaseDataForm.lambda$addFields$3577bca6$1(CaseDataForm.java:1325)
at jdk.internal.reflect.GeneratedMethodAccessor660.invoke(Unknown Source)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at com.vaadin.event.ListenerMethod.receiveEvent(ListenerMethod.java:706)
... 126 common frames omitted
```